### PR TITLE
chore(EA-1974): Modify build workflow for Input APIs to support .NET 8

### PR DESCRIPTION
**Pull request by TurcanStanislav**

### CODE REVIEW

File: .github/workflows/pnp-input-api-build.yml [hunk 1]
```diff
@@ -75,9 +75,36 @@ on:
         type: boolean
         required: false
         default: false
+      dotnet-version:
+        description: ".NET version"
```
Comment (nadiiaboichuk) [inline .github/workflows/pnp-input-api-build.yml:79]: What is the reason why do you switch to double quotes usage?

File: .github/workflows/pnp-input-api-build.yml [hunk 2]
```diff
@@ -374,11 +401,11 @@ jobs:
             Build failed on ${{ github.event_name }} event. Workflow: ${{ github.workflow }}. Job: ${{github.job}}. Run id: <https://github.com/${{ github.repository }}/actions/runs/${{ github.run_id }}|${{ github.run_id }}>
           channel: ${{ secrets.slack-channel }}
           service-account-key: ${{ secrets.SECRET_AUTH }}
-      
+
   release:
     if: github.ref == 'refs/heads/master'
     runs-on: ubuntu-latest
-    needs: acceptance
+    needs: [staging, acceptance]
```
Comment (nadiiaboichuk) [inline .github/workflows/pnp-input-api-build.yml:408]: Why did you add dependency to staging job here?

File: .github/workflows/pnp-input-api-build.yml (hunk 1 above)
Comment (TurcanStanislav) [reply .github/workflows/pnp-input-api-build.yml:79]: This change was done by the yml formatter extension I use in VS Code. Are the problems with this change? Maybe you have maintainability concerns?

File: .github/workflows/pnp-input-api-build.yml (hunk 2 above)
Comment (TurcanStanislav) [reply .github/workflows/pnp-input-api-build.yml:408]: Thought it is a good idea. Actually, the acceptance step already requires staging, so maybe it is not really needed. If so, then should I also remove this for Query API build workflow?

File: .github/workflows/pnp-input-api-build.yml (hunk 1 above)
Comment (nadiiaboichuk) [reply .github/workflows/pnp-input-api-build.yml:79]: Both types of quotes can be used in YAML, however I think I saw only single quotes usage in the github actions documentation, that's why I asked what was the reason for switching to double quotes

File: .github/workflows/pnp-input-api-build.yml (hunk 2 above)
Comment (nadiiaboichuk) [reply .github/workflows/pnp-input-api-build.yml:408]: Yes, you are right, `acceptance` job has already dependency to `staging` job that's why it's not needed for `release` job. In the workflow for Query APIs we might not have acceptance tests that's why we need to have dependency to both `staging` and `acceptance` jobs.

File: .github/workflows/pnp-input-api-build.yml (hunk 1 above)
Comment (TurcanStanislav) [reply .github/workflows/pnp-input-api-build.yml:79]: I have found this issue https://github.com/actions/runner/issues/866 and seems double quotes are not supported in expression language in yml files. But only there.

**Review by nadiiaboichuk — APPROVED** [review]
